internal/rbd/controllerserver.go

```diff
@@ -225,33 +232,36 @@ func (cs *ControllerServer) parseVolCreateRequest(
 }
 
 func buildCreateVolumeResponse(req *csi.CreateVolumeRequest, rbdVol *rbdVolume) *csi.CreateVolumeResponse {
-	// remove kubernetes csi prefixed parameters.
-	volumeContext := k8s.RemoveCSIPrefixedParameters(req.GetParameters())
-	volumeContext["pool"] = rbdVol.Pool
-	volumeContext["journalPool"] = rbdVol.JournalPool
-	volumeContext["imageName"] = rbdVol.RbdImageName
-	if rbdVol.RadosNamespace != "" {
-		volumeContext["radosNamespace"] = rbdVol.RadosNamespace
-	}
-
-	if rbdVol.DataPool != "" {
-		volumeContext["dataPool"] = rbdVol.DataPool
-	}
-
 	volume := &csi.Volume{
 		VolumeId:      rbdVol.VolID,
 		CapacityBytes: rbdVol.VolSize,
-		VolumeContext: volumeContext,
 		ContentSource: req.GetVolumeContentSource(),
 	}
-	if rbdVol.Topology != nil {
-		volume.AccessibleTopology = []*csi.Topology{
+
+	// remove kubernetes csi prefixed parameters.
+	volumeContext := k8s.RemoveCSIPrefixedParameters(req.GetParameters())
+	volumeContext["journalPool"] = rbdVol.JournalPool
+	volumeContext["imageName"] = rbdVol.RbdImageName
+
+	if rbdVol.Topology == nil {
+		volumeContext["pool"] = rbdVol.Pool
```

Comment:
why set these parameters only  when its not `topologyConstraintPools`?
I think, we need the these parameters in either case.

Reply:
It's more like how we wanted,
If we are doing topology provisioning, then pool doesn't make sense, user should look around the topology pools list, still I am fine to update it back if we agree..

Reply:
as you have opened an issue https://github.com/ceph/ceph-csi/issues/4497 to remove the `topologyConstraintPools` parameter, there will be no way to know the `pool` for topology provision.

Also, we need `pool` parameter in other places like `RegenerateJournal`
https://github.com/ceph/ceph-csi/blob/991343d9e5ee5780317df6976e206a864e6b94c0/internal/rbd/rbd_journal.go#L586-L588